readthedocs/proxito/tests/test_full.py

```diff
@@ -185,6 +185,61 @@ def test_custom_robots_txt(self, storage_mock):
             response['x-accel-redirect'], '/proxito/html/project/latest/robots.txt',
         )
 
+    @mock.patch('readthedocs.proxito.views.get_storage_class')
+    def test_directory_indexes(self, storage_mock):
+        self.project.versions.update(active=True, built=True)
+        storage_mock()().exists.return_value = True
+        storage_mock()().open().read.return_value = 'foo'
+        response = self.client.get(
+            reverse('serve_error_404', kwargs={'proxito_path': '/en/latest/index-exists'}),
+            HTTP_HOST='project.readthedocs.io',
+        )
+        self.assertEqual(
+            response.content, b'foo'
+        )
+        self.assertEqual(
+            response.status_code, 200
+        )
+
+    @mock.patch('readthedocs.proxito.views.get_storage_class')
+    def test_404_storage_serves_404(self, storage_mock):
+        self.project.versions.update(active=True, built=True)
+
+        storage_mock()().exists.side_effect = [False, False, True]
+        response = self.client.get(
+            reverse('serve_error_404', kwargs={'proxito_path': '/en/fancy-version/index-exists'}),
```

Comment:
The path is confusing here because the `index.html` does not exist in this case.

```suggestion
            reverse('serve_error_404', kwargs={'proxito_path': '/en/fancy-version/not-found'}),
```